lib/web_ui/lib/src/engine/platform_views/content_manager.dart

```diff
@@ -0,0 +1,185 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of engine;
+
+/// A function which takes a unique `id` and some `params` and creates an HTML element.
+///
+/// This is made available to end-users through dart:ui in web.
+typedef ParameterizedPlatformViewFactory = html.Element Function(
+  int viewId, {
+  Map<dynamic, dynamic>? params,
+});
+/// A function which takes a unique `id` and creates an HTML element.
+///
+/// This is made available to end-users through dart:ui in web.
+typedef PlatformViewFactory = html.Element Function(int viewId);
+
+/// This class handles the lifecycle of Platform Views in the DOM of a Flutter Web App.
+///
+/// There are three important parts of Platform Views that this class manages:
+///
+/// * `factories`: The functions used to render the contents of any given Platform
+/// View by its `viewType`.
+/// * `contents`: The result [html.Element] of calling a `factory` function.
+/// * `slots`: A special HTML tag that will be used by the framework to "reveal"
+/// the `contents` of a Platform View.
+///
+/// This class keeps a registry of `factories`, `slots` and `contents` so the
+/// framework can CRUD Platform Views as needed, regardless of the rendering backend.
+class PlatformViewManager {
+  // The factory functions, indexed by the viewType
+  final Map<String, Function> _factories = {};
+
+  // The references to <slot> tags, indexed by their framework-given ID.
+  final Map<int, html.Element> _slots = {};
```

Comment:
As discussed offline, let's try to remove this cache.

Reply:
This will be cached by the proposed `embedded_views.dart` tuple, instead of the content_manager (since this cache seems to only be required for the Canvaskit renderer)

Reply:
(Slot cache removed from the content_manager.dart)